src/system/sys_profiler.hh

```diff
@@ -574,9 +707,41 @@ public:
      * Check if the instance was created.
      */
     static bool is_initialized() { return (_instance != NULL); }
```

Comment:
Yes, before calling `malloc_notify`a nd `malloc_free` we have to make sure `Profiler` is created and static variable `_instance` is always available. We can't test `_instance` variable because it is private.